nixos/{gnome,pantheon,cinnamon,budgie}: Standardize notExcluded

jtojnar opened the pull request:
Follow up to https://github.com/NixOS/nixpkgs/pull/247036

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jtojnar committed:
nixos/lib: Add disablePackageByName

We do this in multiple DE modules and the behaviour was not consistent.

jtojnar committed:
nixos/gnome: Switch notExcluded to disablePackageByName

For consistency with other DEs.

jtojnar committed:
nixos/cinnamon: Switch notExcluded to disablePackageByName

For consistency with other DEs.

jtojnar committed:
nixos/budgie: Switch notExcluded to disablePackageByName

Previously, this check was inconsistent with `removePackagesByName` – the latter compares the packages using their names.

jtojnar committed:
nixos/pantheon: Switch notExcluded to disablePackageByName

Previously, this check was inconsistent with `removePackagesByName` – the latter compares the packages using their names.

bobby285271 commented:
The `nixosTests.cinnamon` is a good way to test regressions I think as we have `programs.gnome-terminal.enable = mkDefault (notExcluded pkgs.gnome-terminal);` in the module and we have the `Open GNOME Terminal` subtest. But the test does not eval for me yet :-)

@ofborg test cinnamon

bobby285271 commented:
> and looks like it is no longer "notExcluded"

Ugh, to be clear, `nixosTests.cinnamon` now fails with `machine # -bash: line 1: gnome-terminal: command not found`. `notExcluded` really means "not **included** in excludePackages" :joy:

@ofborg test cinnamon

jtojnar commented on nixos/lib/utils.nix:
```diff
@@ -343,6 +343,25 @@ utils = rec {
+    let
+      namesToDisable = map getName packagesToDisable;
+    in
+      elem (getName package) namesToDisable;
```
Ugh, should be:

```suggestion
      !elem (getName package) namesToDisable;
```

I really need find time to test it properly.

jtojnar commented on nixos/lib/utils.nix:
```diff
@@ -343,6 +343,25 @@ utils = rec {
+       enableByPackageName nautilus [ file-roller totem ]
+       => true
+  */
+  enableByPackageName = package: packagesToDisable:
```
Also the name is probably too confusing.

jtojnar commented:
Fixed the negation and renamed the function to `disablePackageByName`.

Now `nix-build -A nixosTests.gnome -A nixosTests.cinnamon -A nixosTests.pantheon -A nixosTests.budgie` pass successfully.
